### ISSUE TITLE
Duplicated Floppy-Disks

### ISSUE THREAD

**Issue by SeaLife** [SquidDev-CC/CC-Tweaked#22]
Hey!,

i encountered a duplicating bug while using the latest CC:Tweaked within FTB Direwolf 20 ModPack.

Sometimes (after a Reboot of the MC-Server for example) the computers freezes (you cant interact with them) - i then tried the `/computercraft shutdown` and `/computer turn-on` command to reboot the computers but if they stay in this state i can duplicate disks by putting them into a disk drive (which is connected to the computer all the time).

If this happens, there is a stacktrace in the console:

```
java.util.concurrent.ExecutionException: java.lang.IllegalStateException: File system has not been created
        at java.util.concurrent.FutureTask.report(FutureTask.java:122) ~[?:1.8.0_111]
        at java.util.concurrent.FutureTask.get(FutureTask.java:192) ~[?:1.8.0_111]
        at net.minecraft.util.Util.func_181617_a(SourceFile:47) [h.class:?]
        at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:721) [MinecraftServer.class:?]
        at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:396) [nz.class:?]
        at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:666) [MinecraftServer.class:?]
        at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:524) [MinecraftServer.class:?]
        at java.lang.Thread.run(Thread.java:745) [?:1.8.0_111]
Caused by: java.lang.IllegalStateException: File system has not been created
        at dan200.computercraft.core.apis.ComputerAccess.mountWritable(ComputerAccess.java:85) ~[ComputerAccess.class:?]
        at dan200.computercraft.core.apis.PeripheralAPI$PeripheralWrapper.mountWritable(PeripheralAPI.java:140) ~[PeripheralAPI$PeripheralWrapper.class:?]
        at dan200.computercraft.core.apis.ComputerAccess.mountWritable(ComputerAccess.java:74) ~[ComputerAccess.class:?]
        at dan200.computercraft.shared.peripheral.diskdrive.TileDiskDrive.mountDisk(TileDiskDrive.java:532) ~[TileDiskDrive.class:?]
        at dan200.computercraft.shared.peripheral.diskdrive.TileDiskDrive.func_70299_a(TileDiskDrive.java:322) ~[TileDiskDrive.class:?]
        at dan200.computercraft.shared.peripheral.diskdrive.TileDiskDrive.onActivate(TileDiskDrive.java:105) ~[TileDiskDrive.class:?]
        at dan200.computercraft.shared.common.BlockGeneric.func_180639_a(BlockGeneric.java:143) ~[BlockGeneric.class:?]
        at net.minecraft.server.management.PlayerInteractionManager.func_187251_a(PlayerInteractionManager.java:446) ~[or.class:?]
        at net.minecraft.network.NetHandlerPlayServer.func_184337_a(NetHandlerPlayServer.java:739) ~[pa.class:?]
        at net.minecraft.network.play.client.CPacketPlayerTryUseItemOnBlock.func_148833_a(SourceFile:55) ~[ma.class:?]
        at net.minecraft.network.play.client.CPacketPlayerTryUseItemOnBlock.func_148833_a(SourceFile:11) ~[ma.class:?]
        at net.minecraft.network.PacketThreadUtil$1.run(SourceFile:13) ~[hv$1.class:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:1.8.0_111]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:1.8.0_111]
        at net.minecraft.util.Util.func_181617_a(SourceFile:46) ~[h.class:?]
        ... 5 more
```

Hopefully this is enough to find the error - i can provide more information is neccessary :)

Regards
SeaLife

**Comment by SquidDev** [comment]
Thanks for the report!

> Sometimes (after a Reboot of the MC-Server for example) the computers freezes (you cant interact with them)

I've not seen this before, and I suspect it's the root cause of the bug. Would it be possible to turn `logPeripheralErrors` on in the config and provide the full log next time this happens? This will hopefully allow me to track down what's broken.

**Comment by SeaLife** [comment]
After some time of testing i cant reproduce this issue. But i changed something in the "environment". While i encountered the issue i was using the IC2 Chunk Loader for chunk loading. Now i've claimed the chunk and loaded it via FTB Utilities and it works... (got some issue here and there but overall it worked)...

Regards
SeaLife

**Comment by SquidDev** [comment]
OK, I'm going to close the issue for now. Do comment if it reoccurs though

**Comment by SeaLife** [comment]
Okay - after a "emergency" reboot due to a networking incident brought this issue back... (but as far as i know only on one computer+floppyDisk)

```
java.util.concurrent.ExecutionException: java.lang.IllegalStateException: File system has not been created
        at java.util.concurrent.FutureTask.report(FutureTask.java:122) ~[?:1.8.0_111]
        at java.util.concurrent.FutureTask.get(FutureTask.java:192) ~[?:1.8.0_111]
        at net.minecraft.util.Util.func_181617_a(SourceFile:47) [h.class:?]
        at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:721) [MinecraftServer.class:?]
        at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:396) [nz.class:?]
        at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:666) [MinecraftServer.class:?]
        at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:524) [MinecraftServer.class:?]
        at java.lang.Thread.run(Thread.java:745) [?:1.8.0_111]
Caused by: java.lang.IllegalStateException: File system has not been created
        at dan200.computercraft.core.apis.ComputerAccess.mountWritable(ComputerAccess.java:85) ~[ComputerAccess.class:?]
        at dan200.computercraft.core.apis.PeripheralAPI$PeripheralWrapper.mountWritable(PeripheralAPI.java:140) ~[PeripheralAPI$PeripheralWrapper.class:?]
        at dan200.computercraft.core.apis.ComputerAccess.mountWritable(ComputerAccess.java:74) ~[ComputerAccess.class:?]
        at dan200.computercraft.shared.peripheral.diskdrive.TileDiskDrive.mountDisk(TileDiskDrive.java:532) ~[TileDiskDrive.class:?]
        at dan200.computercraft.shared.peripheral.diskdrive.TileDiskDrive.func_70299_a(TileDiskDrive.java:322) ~[TileDiskDrive.class:?]
        at net.minecraft.inventory.Slot.func_75215_d(Slot.java:70) ~[agr.class:?]
        at net.minecraft.inventory.Container.func_75135_a(Container.java:659) ~[afr.class:?]
        at dan200.computercraft.shared.peripheral.diskdrive.ContainerDiskDrive.func_82846_b(ContainerDiskDrive.java:63) ~[ContainerDiskDrive.class:?]
        at net.minecraft.inventory.Container.func_184996_a(Container.java:246) ~[afr.class:?]
        at net.minecraft.network.NetHandlerPlayServer.func_147351_a(NetHandlerPlayServer.java:1158) ~[pa.class:?]
        at net.minecraft.network.play.client.CPacketClickWindow.func_148833_a(CPacketClickWindow.java:38) ~[lf.class:?]
        at net.minecraft.network.play.client.CPacketClickWindow.func_148833_a(CPacketClickWindow.java:12) ~[lf.class:?]
        at net.minecraft.network.PacketThreadUtil$1.run(SourceFile:13) ~[hv$1.class:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:1.8.0_111]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:1.8.0_111]
        at net.minecraft.util.Util.func_181617_a(SourceFile:46) ~[h.class:?]
        ... 5 more
```

Only breaking and rebuilding the disk-drive fixed the issue, no more log entries there even with the debug option enabled.

Regards
SeaLife

**Comment by SeaLife** [comment]
Hey,

i've found out how to reproduce...:

So how i can duplicate disks without any server restart....

- Build a Computer
- Attach a Disk-Drive to it
- Boot the Computer
- Shutdown the Computer via STRG+S (CTRL+S)
- Shift+Rightclick disk to duplicate into the disk-drive
- Voila, one disk in hand, one in the drive (metadata equals)

Exception-Stacktrace doesnt changed to what i've reported... :frowning:

**Comment by SquidDev** [comment]
For reference, [SeaLife put together this video](https://www.youtube.com/watch?v=Eycmu94a7u0&feature=youtu.be). I've followed the steps, but am unable to reproduce on my end. I suspect there is another mod interfering with this, but I'm not sure how.

**Comment by vpontin** [comment]
Yeah there a LOT of mods on his end, even web radio O.o
I never reached the point to make floppies on my survival worlds, but i can try to test this here when i have free time.

**Comment by SeaLife** [comment]
No Web Radios, there is Spotify running in the background :D

For full reference, im using all of these mods: https://mc.r3ktm8.de/?mod=FTBPresentsDirewolf20112

Regards
SeaLife

**Comment by JasonTheKitten** [comment]
Did you try dragging your mods folder into a different location, making a new folder in it's place, and then dragging computer-craft back in?

**Comment by SeaLife** [comment]
Apparently i did not tried only using CC:Tweaked :/

Regards
SeaLife

**Comment by shelbykauth** [comment]
> Sometimes (after a Reboot of the MC-Server for example) the computers freezes (you cant interact with them)

I believe this is maybe due to the bug I reported the other day.  https://github.com/SquidDev-CC/CC-Tweaked/issues/34
Are you using Peripherals Plus One Build T58?  If so, update it to Build T61, and see if your problem goes away.  I was able to reproduce the computers freezing quite regularly by restarting my entire singleplayer client (and thus world as well).  If I left the world and went back in, the problem would vanish, likewise if I went sufficiently far away so the chunks reloaded when I came back.  With T61, the freezing bug became unreproduceable.

Following the steps provided here:
> Build a Computer
> Attach a Disk-Drive to it
> Boot the Computer
> Shutdown the Computer via STRG+S (CTRL+S)
> Shift+Rightclick disk to duplicate into the disk-drive
> Voila, one disk in hand, one in the drive (metadata equals)

I thought it didn't work at first.  But then I pulled the one disk out and got two disks instead.  Every time I remove the disk, escape, right click on the disk drive again, the disk is back in the drive.
- Turning on the computer removes the bug.
- Forcing shutdown of the computer makes the bug come back on.
- Normal shutdown also causes the bug.  As does removing the computer, so that makes debugging harder, as in knowing where the bug starts exactly.
- Making a new unattached disk drive, the bug does not exist.  It continues to not exist when you place a computer near it.
- Turn the computer on and then turn it off or break it, bug exists.
- Multiple computers will cause error if just one of them is off or broken.
- My method of checking to see if the bug is still there is holding the disk in my cursor, going to the drive, and left clicking multiple times (to put it in and take it out, and seeing when the disk duplicates.

Btw, I have JEI.  I do have cheats on.  But I did turn the JEI cheatmode off for testing.  Bug persisted, so I turned it back on.
